src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -48,6 +49,49 @@ class FileService(
   def getFiles(consignmentId: UUID): Future[Files] = {
     fileRepository.getFilesWithPassedAntivirus(consignmentId).map(r => Files(r.map(_.fileid)))
   }
+
+  def getFileMetadata(consignmentId: UUID): Future[List[File]] = {
```

Comment:
This is an interesting one. The method is now fetching a lot of different data, so we should probably use deferred resolvers, but it's already inside another deferred resolver and I've never tried nesting deferred resolvers in Sangria.

It looks like you _can_ do it, though: https://github.com/sangria-graphql/sangria/issues/63

I don't think it's going to be trivial to rewrite this code with nested resolvers, but we should do it because this is going to get even more complicated when we add the antivirus metadata. So I think I'll add a refactoring card for us to dig into this properly, and stick with what you've got here. How does that sound?

Reply:
I've made a card about it: https://national-archives.atlassian.net/browse/TDR-948